File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/DSRestParameter.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "DSRestParameter.h"


@implementation DSRestParameter

@synthesize name;
@synthesize direction;
@synthesize typeName;
@synthesize DBXType;

- (id) init{
	self = [super init]; 
	if (self) {
	  value = [[DBXWritableValue alloc]init];	
   }
	return self;
}
- (id) initWithParameter:(NSString *) aname  withDirection: (DSRESTParamDirection) adirection 
			withTypeName: (NSString *) aTypeName {
	self = [self init];
	if (self) {
		name = [NSString stringWithString:aname];
		direction = adirection;
		typeName = [NSString stringWithString:aTypeName];
		
	}
	return self;
	
}

-(DBXValue*) getValue {
	return value  ;
}


@end

File: Demos/Datasnap/Server/proxy/freepascal_ios50/DSRESTParameter.pas
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2012 Embarcadero Technologies, Inc.
//
//*******************************************************

unit DSRestParameter;

interface
  uses
    DSRestTypes,DBXValue;
  type
   TDSRestParameter = class
   private
    FName:string;
    FDirection:TDSRESTParamDirection;
    FDBXType:TDBXDataTypes;
    FTypeName:String;
    FValue:TDBXValue;
   public
     constructor create; overload;
     constructor create(aName:string;aDirection:TDSRESTParamDirection;
     aTypeName:STring); overload;
     destructor Destroy; override;

     function isDBXValue:boolean;
     property  Name:string read FName write FName;
     property  Direction:TDSRESTParamDirection read FDirection write FDirection;
     property  DBXType:TDBXDataTypes read FDBXType write FDBXType;
     property  TypeName:String read FTypeName write FTypeName;
     property  Value:TDBXValue read FValue write FValue;


   end;

implementation
uses sysutils;

{ TDSRestParameter }
constructor TDSRestParameter.create;
begin
  inherited create;
  FValue:= TDBXWritableValue.Create;
end;

constructor TDSRestParameter.create(aName: string;
  aDirection: TDSRESTParamDirection; aTypeName: STring);
begin
  self.create;
  FName:= aName;
  FDirection:= aDirection;
  FTypeName:= aTypeName;
end;

destructor TDSRestParameter.Destroy;
begin
  FValue.Free;
  inherited;
end;

function TDSRestParameter.isDBXValue:Boolean;
begin
 result := TypeIsDBXValue(TypeName);
end;
end.
